refactor: Fix code according to clippy suggestions

magurotuna opened the pull request:
~~This PR adds new CI steps where rustfmt and clippy check if the code is formatted and linted. Generated codes are ignored.~~

This fixes the code according to clippy suggestions, including renaming `into_node` method of `NodeTrait` to `as_node`, which will be a breaking change.

magurotuna commented on rs-lib/src/types.rs:
```diff
@@ -264,7 +264,12 @@ where
   fn parent(&self) -> Option<Node<'a>>;
   fn children(&self) -> Vec<Node<'a>>;
+
+  // TODO: according to the convention, the name should preferably be `as_node`
+  // https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv
+  #[allow(clippy::wrong_self_convention)]
   fn into_node(&self) -> Node<'a>;
```
As commented, the name `into_node` doesn't follow the naming convention. Personally I feel like it should be renamed to `as_node`, it'll be a breaking change though. Any thoughts?

dsherret commented:
Hey @magurotuna, thanks for this! I think these changes are good, but I don't want to bother with linting and formatting on the CI since there are not a lot of people maintaining this project. If there were a lot of maintainers then yes this would be great.

For linting, I've found that the vast majority of linting errors don't matter and it only causes me an annoyance when the build fails over something super trivial. Ex. in #19 the source of the tokens from swc is a vector, so although it's nicer for the API to use `&[TokenAndSpan]`, it's not really necessary. The one catch in this PR it has helped with is `into_node`, which definitely should be `as_node`; however, I don't think it's worth all the other trouble linters bring. The one problem with not having linting here is that linting errors in this library can cause downstream linting errors in projects like deno_lint that do use clippy... that said, I think these scenarios are rare and I'll just run clippy locally every now and again to try to catch them.

For formatting, it's not a big deal if parts of the source code don't get formatted since there aren't a lot of people working on this and I'd rather not fail the build. By the way, right now rustfmt is handled by dprint...

https://github.com/dprint/dprint-swc-ecma-ast-view/blob/1b0e0ba18c4f00eae1616cc541117d2bf5a8a8e5/dprint.json#L26

...so if we were to add this, then we should use dprint/check (https://github.com/dprint/check).

So, all that said, **it would be great if you could check in these changes** except for the ci.yml change and then also do the breaking `into_node` to `as_node` change. Thanks!

magurotuna commented:
@dsherret Thanks for the feedback!
Even if there are a few people taking part in this project, I'd prefer to set up the CI to check for format and lint. Nevertheless, what you are saying also sounds reasonable to me to some extent, so I reverted the CI setup and renamed `into_node` to `as_node` :) 
Could you please take a look again?

dsherret commented:
@magurotuna awesome. Thanks! 🙂 Will be released in 0.18.0 once the CI completes.
